src/mahoji/lib/abstracted_commands/nightmareZoneCommand.ts

```diff
@@ -0,0 +1,482 @@
+		aliases: ['flax']
+	},
+	{
+		name: 'Bucket of sand',
+		output: new Bank().add('Bucket of sand', 1),
+		cost: 200,
+		aliases: ['sand']
+	},
+	{
+		name: 'Seaweed',
+		output: new Bank().add('Seaweed', 1),
+		cost: 200,
+		aliases: []
+	},
+	{
+		name: 'Dragon scale dust',
+		output: new Bank().add('Dragon scale dust', 1),
+		cost: 750,
+		aliases: ['dragon dust']
+	},
+	{
+		name: 'Compost potion(4)',
+		output: new Bank().add('Compost potion(4)', 1),
+		cost: 5000,
+		aliases: ['compost potion']
+	},
+	{
+		name: 'Air rune',
+		output: new Bank().add('Air rune', 1),
+		cost: 25,
+		aliases: []
+	},
+	{
+		name: 'Water rune',
+		output: new Bank().add('Water rune', 1),
+		cost: 25,
+		aliases: []
+	},
+	{
+		name: 'Earth rune',
+		output: new Bank().add('Earth rune', 1),
+		cost: 25,
+		aliases: []
+	},
+	{
+		name: 'Fire rune',
+		output: new Bank().add('Fire rune', 1),
+		cost: 25,
+		aliases: []
+	},
+	{
+		name: 'Pure essence',
+		output: new Bank().add('Pure essence', 1),
+		cost: 70,
+		aliases: []
+	},
+	/* TODO: Lock this to max 15 Herb boxes per day
+	{
+		name: 'Herb box',
+		output: new Bank().add('Herb box', 1),
+		cost: 9500,
+		aliases: []
+	},
+	*/
+	{
+		name: 'Vial of water',
+		output: new Bank().add('Vial of water', 1),
+		cost: 145,
+		aliases: []
+	}
+];
+
+export async function nightmareZoneStatsCommand(user: MUser) {
+	const scores = await getMinigameEntity(user.id);
+	return `**Nightmare Zone Stats:**
+
+**Nightmare Zone monsters killed:** ${scores.nmz}.
+**Nightmare Zone points:** ${user.user.nmz_points} Points.`;
+}
+
+export async function nightmareZoneStartCommand(user: MUser, strategy: NMZStrategy, channelID: string) {
+	const skillReqs: Skills = {
+		defence: 70,
+		strength: 70,
+		attack: 70,
+		hitpoints: 70,
+		prayer: 43
+	};
+
+	const [hasReqs] = hasSkillReqs(user, skillReqs);
+	if (!hasReqs) {
+		return `You not meet skill requirements, you need ${Object.entries(skillReqs)
+			.map(([name, lvl]) => `${lvl} ${name}`)
+			.join(', ')}.`;
+	}
+
+	const { QP } = user;
+
+	if (QP < 100) {
+		return `The Nightmare Zone minigame requires **100 QP**, and you have ${QP} QP.\n`;
+	}
+
+	// Check if user have full dharok's
+	if (
+		!user.hasEquippedOrInBank(
+			["Dharok's helm", "Dharok's platebody", "Dharok's platelegs", "Dharok's greataxe"],
+			'every'
+		)
+	) {
+		return "The Nightmare Zone minigame requires full Dharok's equipment for optimal experience/points gained.";
+	}
+
+	const boosts = [];
+
+	let timePerMonster = Time.Minute * 2;
+
+	// combat stat boosts
+	// No monster ID so use -1 as placeholder
+	const [, , attackStyles] = resolveAttackStyles(user, { monsterID: -1 });
+	const skillTotal = sumArr(attackStyles.map(s => user.skillLevel(s))) + user.skillLevel(SkillsEnum.Hitpoints);
+	if (attackStyles.includes(SkillsEnum.Ranged) || attackStyles.includes(SkillsEnum.Magic)) {
+		return 'The Nightmare Zone minigame requires melee combat for efficiency, swap training style using `/minion train style:`';
```

Comment:
We could add buttons for switching attack styles, so the user can just click a button here, but you dont have to do that.